File: src/DFS.java
public class DFS {
    char[][] graph;
    boolean[][] visited;
    Bike start;
    int counter = 0;

    public DFS(char[][] graph, Bike start) {
        this.graph = graph;
        visited = new boolean[graph.length][graph[0].length];
        this.start = start;
    }

    int dfs() {
//        for (int i = 0; i < graph.length; i++) {
//            System.out.print('[');
//            for (int j = 0; j < graph[0].length; j++) {
//                System.out.print("'" + graph[i][j] + (j == graph[0].length - 1 ? "'" : "',"));
//            }
//            System.out.println("],");
//        }
//        System.out.println("START VALUE IS " + start.row + " " + start.col);
        dfs_driver(start);
//        System.out.println("COUNTER IS " + counter);
        return counter;
    }

    void dfs_driver(Bike point) {
        if (point.row < 1 || point.col < 1 || point.row >= 16 || point.col >= 17)
            return;
        if (visited[point.row][point.col])
            return;
        char c = graph[point.row][point.col];

        if (c != '.') return;
        counter++;

        visited[point.row][point.col] = true;

        dfs_driver(new Bike(point.row - 1, point.col));
        dfs_driver(new Bike(point.row + 1, point.col));
        dfs_driver(new Bike(point.row, point.col - 1));
        dfs_driver(new Bike(point.row, point.col + 1));
    }
}

File: server/dfs.js
class Bike {
  constructor(row, col) {
    this.row = row;
    this.col = col;
  }
}

let visited = [];
let counter = 0;

const start_dfs = (graph, start) => {
  // reset visited
  counter = 0;
  for (var i = 0; i < 17; i++) {
    let temp = [];
    for (var j = 0; j < 17; j++) {
      temp.push(false);
    }
    visited.push(temp);
  }

  dfs(graph, start);
  return counter;
};

const dfs = (graph, start) => {
  console.log(start);
  if (start.row < 1 || start.col < 1 || start.row >= 16 || start.col >= 17)
    return;
  if (visited[start.row][start.col]) return;
  let c = graph[start.row][start.col];

  if (c != ".") return;
  counter++;

  visited[start.row][start.col] = true;

  dfs(graph, new Bike(start.row - 1, start.col));
  dfs(graph, new Bike(start.row + 1, start.col));
  dfs(graph, new Bike(start.row, start.col - 1));
  dfs(graph, new Bike(start.row, start.col + 1));
};

module.exports = start_dfs;
